pkg/debugapi/chequebook.go

```diff
@@ -244,6 +244,50 @@ func (s *Service) swapCashoutHandler(w http.ResponseWriter, r *http.Request) {
 	jsonhttp.OK(w, swapCashoutResponse{TransactionHash: txHash.String()})
 }
 
+func (s *Service) swapReCashoutHandler(w http.ResponseWriter, r *http.Request) {
+	addr := mux.Vars(r)["peer"]
+	peer, err := swarm.ParseHexAddress(addr)
+	if err != nil {
+		s.logger.Debugf("debug api: cashout peer: invalid peer address %s: %v", addr, err)
+		s.logger.Errorf("debug api: cashout peer: invalid peer address %s", addr)
+		jsonhttp.NotFound(w, errInvalidAddress)
+		return
+	}
+
+	ctx := r.Context()
+	if price, ok := r.Header[gasPriceHeader]; ok {
+		p, ok := big.NewInt(0).SetString(price[0], 10)
+		if !ok {
+			s.logger.Error("debug api: cashout peer: bad gas price")
+			jsonhttp.BadRequest(w, errBadGasPrice)
+			return
+		}
+		ctx = sctx.SetGasPrice(ctx, p)
+	}
+
+	if limit, ok := r.Header[gasLimitHeader]; ok {
+		l, err := strconv.ParseUint(limit[0], 10, 64)
+		if err != nil {
+			s.logger.Debugf("debug api: cashout peer: bad gas limit: %v", err)
+			s.logger.Error("debug api: cashout peer: bad gas limit")
+			jsonhttp.BadRequest(w, errBadGasLimit)
+			return
+		}
+		ctx = sctx.SetGasLimit(ctx, l)
+	}
```

Comment:
should we do nothing if `gasLimit` is empty?

Reply:
`gasLimit` will be set in `pepraeTransaction` function if use not post.